### DISCUSSION TITLE
Headless API user login

### DISCUSSION THREAD

**Discussion by MichelJonkman** [statamic/cms#6429] [Q&A] [answered]
I'm looking to build a Nuxt application with Statamic as back-end coming from Strapi but my only concern is: is it easy enough to implement a system that allows people to register and access restricted content?

**Comment by joshuablum** [comment, marked as answer]
Hey @99michel99,

From what I understand there are multiple ways you could achieve this.

First a heads-up: Statamic's APIs (REST and GraphQL) don't offer built-in authentication right now. See [here](https://statamic.dev/rest-api#authentication) and [here](https://statamic.dev/graphql#authorization).

You could do your auth either on the backend (Statamic/Laravel) or a more frontend heavy approach in your Nuxt application. Depends on your preference.

With Laravel it's quite easy to come up with your own solution by leveraging it's great ecosystem of (first-party) packages, like [Laravel Sanctum](https://laravel.com/docs/9.x/sanctum) or [Laravel Passport](https://laravel.com/docs/9.x/passport). Sanctum is a more lightweight solution and Passport a full-fledged OAuth2 server.

There is also the Nuxt Auth module that offers support for [Laravel Sanctum](https://auth.nuxtjs.org/providers/laravel-sanctum), [Laravel Passport](https://auth.nuxtjs.org/providers/laravel-passport) and [Laravel JWT](https://auth.nuxtjs.org/providers/laravel-jwt/). You could even use an external auth provider like [Auth0](https://auth.nuxtjs.org/providers/auth0).

Also have a look at the awesome [Memberbox addon](https://statamic.com/addons/jacksleight/memberbox) by @jacksleight. Don't know of how much use it is to you in your use case though but maybe it gives you a head start.

Hope that helps 🚀

**Reply by MichelJonkman** [reply]
Thank you for the help! Sorry for the late response, I completely forgot about this post... I will take a look at what you've posted whenever I'll go and try it out. Thank you very much!
